src/modules/75-cf/components/SetUpGitSync/__tests__/SetUpGitSync.test.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { render, RenderResult, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import { TestWrapper } from '@common/utils/testUtils'
+import * as cfServices from 'services/cf'
+// import mockGitSync from '@cf/utils/testData/data/mockGitSync'
+import { SetUpGitSync } from '../SetUpGitSync'
+
+// jest.mock('services/cf')
+
+// jest.mock('@cf/hooks/useEnvironmentSelectV2', () => ({
+//   useEnvironmentSelectV2: jest.fn().mockReturnValue({ data: [], loading: false })
+// }))
+
+const renderComponent = (props?: any): RenderResult => {
+  return render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
+      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+      defaultFeatureFlagValues={{
+        FF_FLAG_SYNC_THROUGH_GITEX_ENABLED: props?.gitExFlag
+      }}
+    >
+      <SetUpGitSync />
+    </TestWrapper>
+  )
+}
+
+describe('SetUpGitSync', () => {
+  const createGitRepoMock = jest.fn()
+  beforeEach(() => {
+    // jest.clearAllMocks()
+    jest.spyOn(cfServices, 'useCreateGitRepo').mockReturnValue({
+      cancel: jest.fn(),
+      loading: false,
+      error: null,
+      mutate: createGitRepoMock
+    })
+  })
+
+  afterEach(() => {
+    // jest.clearAllMocks()
+    jest.resetAllMocks()
+  })
+
+  test('it should render correctly', async () => {
+    renderComponent({ gitExFlag: true })
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'cf.featureFlags.setupGitSync' })).toBeVisible())
+  })
+
+  test('It should redirect to Git Management on click of "Set Up Git Sync" button when GitEx flag is false', async () => {
+    renderComponent({ gitExFlag: false })
+
+    const setupGitBtn = screen.getByRole('button', { name: 'cf.featureFlags.setupGitSync' })
+
+    await waitFor(() => {
+      expect(screen.queryByText('cf.gitSync.setUpGitConnection')).not.toBeInTheDocument()
+      expect(setupGitBtn).toBeVisible()
+    })
```

Comment:
Same here, you rarely need to wait for initial render
```suggestion
    expect(screen.queryByText('cf.gitSync.setUpGitConnection')).not.toBeInTheDocument()
    expect(setupGitBtn).toBeVisible()
```